connector_magento/components/backend_adapter.py

```diff
@@ -153,9 +155,19 @@ def call(self, method, arguments, http_method=None, storeview=None):
             # record(method, arguments, result)
             return result
         except (socket.gaierror, socket.error, socket.timeout) as err:
-            raise NetworkRetryableError(
-                'A network error caused the failure of the job: '
-                '%s' % err)
+            if err.response is not None:
+                try:
+                    json = err.response.json()
+                except JSONDecodeError:
+                    raise
```

Comment:
In this case, you need to end up with a network error. What you can do is raise the `MagentoError` before you catch the `JSONDecodeError`. You can just `pass` that exception and then the network exception further below kicks in. That is, if you remove the `else` in line 166. It is redundant anyway, because if an exception is raised earlier we don't end up there. We will always want to raise one of them at this point.

Reply:
done